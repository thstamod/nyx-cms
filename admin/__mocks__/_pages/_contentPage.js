const data = {
  documentTypes: [
    {
      _id: '5e765ba31427b934d26c920d',
      name: 'Home',
      descendants: null,
      __typename: 'DocumentType',
    },
    {
      _id: '5e765bca1427b934d26c920e',
      name: 'About',
      descendants: null,
      __typename: 'DocumentType',
    },
    {
      _id: '5e7b8a88bb9b7822b5a088b7',
      name: 'About2',
      descendants: null,
      __typename: 'DocumentType',
    },
    {
      _id: '5e7b8a9ebb9b7822b5a088b8',
      name: 'About3',
      descendants: null,
      __typename: 'DocumentType',
    },
    {
      _id: '5e7b8ba9bb9b7822b5a088b9',
      name: 'Test',
      descendants: null,
      __typename: 'DocumentType',
    },
    {
      _id: '5e7b8fb9529e583810ff85fb',
      name: 'Test',
      descendants: [
        {
          documentType: null,
          __typename: 'descendants',
        },
        {
          documentType: null,
          __typename: 'descendants',
        },
      ],
      __typename: 'DocumentType',
    },
    {
      _id: '5e7b8feb9f4b4e3916cc6a57',
      name: 'Test',
      descendants: [
        {
          documentType: null,
          __typename: 'descendants',
        },
        {
          documentType: null,
          __typename: 'descendants',
        },
      ],
      __typename: 'DocumentType',
    },
    {
      _id: '5e7b8fed9f4b4e3916cc6a58',
      name: 'Test',
      descendants: [
        {
          documentType: null,
          __typename: 'descendants',
        },
        {
          documentType: null,
          __typename: 'descendants',
        },
      ],
      __typename: 'DocumentType',
    },
    {
      _id: '5e7b903d8b93ca3979e56e94',
      name: 'Test',
      descendants: [
        {
          documentType: {
            name: 'Home',
            _id: '5e765ba31427b934d26c920d',
            __typename: 'DocumentType',
          },
          __typename: 'descendants',
        },
        {
          documentType: {
            name: 'About',
            _id: '5e765bca1427b934d26c920e',
            __typename: 'DocumentType',
          },
          __typename: 'descendants',
        },
      ],
      __typename: 'DocumentType',
    },
    {
      _id: '5e7b907e369cbc3a39a5599c',
      name: 'Test',
      descendants: [
        {
          documentType: {
            name: 'Home',
            _id: '5e765ba31427b934d26c920d',
            __typename: 'DocumentType',
          },
          __typename: 'descendants',
        },
        {
          documentType: {
            name: 'About',
            _id: '5e765bca1427b934d26c920e',
            __typename: 'DocumentType',
          },
          __typename: 'descendants',
        },
      ],
      __typename: 'DocumentType',
    },
    {
      _id: '5e7b915f369cbc3a39a5599d',
      name: 'Test',
      descendants: [
        {
          documentType: {
            name: 'Home',
            _id: '5e765ba31427b934d26c920d',
            __typename: 'DocumentType',
          },
          __typename: 'descendants',
        },
        {
          documentType: {
            name: 'About',
            _id: '5e765bca1427b934d26c920e',
            __typename: 'DocumentType',
          },
          __typename: 'descendants',
        },
      ],
      __typename: 'DocumentType',
    },
  ],
};

// eslint-disable-next-line import/prefer-default-export
export { data };
