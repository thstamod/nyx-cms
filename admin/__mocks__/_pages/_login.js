export default {
  data: {
    login: {
      token:
        'eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiI1ZTcxNDczZjE4MTIxYzJmZjI0YzQwZWYiLCJlbWFpbCI6InRlc3RAbWFpbC5jb20iLCJpYXQiOjE1ODc4OTc3NTEsImV4cCI6MTU4NzkwMTM1MX0.3oIAITE622eRvD85ux1t5n1rLAhOJmyc7c7tTMXYgCA',
      tokenExpiration: 1,
    },
  },
};
