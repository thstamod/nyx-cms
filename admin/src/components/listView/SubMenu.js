import React from 'react';
import propTypes from 'prop-types';
import styled from 'styled-components';

const Test = styled('div')`
  font-size: 14px;
  padding-left: 10px;
`;

const SubMenu = ({ data }) => {
  const showSubmenu = (d) => {
    // eslint-disable-next-line arrow-body-style
    const list = d.map(({ documentType }) => {
      console.log(documentType);
      return (
        documentType && <Test key={documentType._id}>{documentType.name}</Test>
      );
    });
    return list;
  };

  return <div>{showSubmenu(data)}</div>;
};

SubMenu.propTypes = {
  data: propTypes.array,
};

export default SubMenu;
