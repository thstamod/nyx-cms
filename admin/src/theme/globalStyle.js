import { createGlobalStyle } from 'styled-components';

const GlobalStyle = createGlobalStyle`
  html,body {
    width:100%;
  height: 100vh;
  display: flex;
  font-family: Karla, serif;
  }
#root {
  width:100%;
  height:100%;
}

  h1,h2,h3,h4,h5 {
    font-family: Rubik, serif;
  }
`;

export default GlobalStyle;
