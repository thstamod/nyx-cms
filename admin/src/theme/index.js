const theme = {
  colors: {
    powderWhite: '#F2F3F4',
    foggyOrange: '#D6804D',
    lightPurple: '#B886A5',
    darkOrange: '#AB5C49',
    darkGrey: '#262728',
  },
  fonts: ['Karla,sans-serif', 'Rubik,sans-serif'],
  fontSizes: {
    small: '1em',
    medium: '2em',
    large: '3em',
  },
};

export default theme;
