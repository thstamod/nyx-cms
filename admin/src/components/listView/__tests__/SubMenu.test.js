import React from 'react';
import '@testing-library/jest-dom/extend-expect';
import { render, cleanup } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import theme from '../../../theme';
import SubMenu from '../SubMenu';
import { data } from '../../../../__mocks__/_pages/_contentPage';

afterEach(cleanup);

test('SubMenu renders', () => {
  const { asFragment } = render(
    <ThemeProvider theme={theme}>
      <SubMenu data={data.documentTypes[10].descendants} />
    </ThemeProvider>
  );

  expect(asFragment()).toMatchSnapshot();
});
