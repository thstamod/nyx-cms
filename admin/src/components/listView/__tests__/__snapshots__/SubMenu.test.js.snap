// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SubMenu renders 1`] = `
<DocumentFragment>
  <div>
    <div
      class="sc-AxjAm huDVTX"
    >
      Home
    </div>
    <div
      class="sc-AxjAm huDVTX"
    >
      About
    </div>
  </div>
</DocumentFragment>
`;
